[CI] MixedClusterClientYamlTestSuiteIT test {p0=indices.create/20_synthetic_source/stored field under object with store_array_source} failing

**Build Scans:**
- [elasticsearch-intake #9183 / 8.15.1_bwc-snapshots](https://gradle-enterprise.elastic.co/s/xauyauxnhay3q)
- [elasticsearch-pull-request #29252 / 8.15.1_bwc-snapshots](https://gradle-enterprise.elastic.co/s/yb3caz7lbcaoe)
- [elasticsearch-pull-request #29149 / 8.15.1_bwc-snapshots](https://gradle-enterprise.elastic.co/s/jnawafyebh7fc)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v8.15.1#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=indices.create/20_synthetic_source/stored field under object with store_array_source}" -Dtests.seed=EA10560D29FDD45E -Dtests.bwc=true -Dtests.locale=ar-SA -Dtests.timezone=Asia/Urumqi -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Bp0%3Dindices.create%2F20_synthetic_source%2Fstored%20field%20under%20object%20with%20store_array_source%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [indices.create/20_synthetic_source:1390]: field [hits.hits.1._source.obj] should be null but was [{foo=hello_from_the_other_side}] expected null, but was:<{foo=hello_from_the_other_side}>
```

**Issue Reasons:**
- [main] 3 failures in test test {p0=indices.create/20_synthetic_source/stored field under object with store_array_source} (42.9% fail rate in 7 executions)
- [main] 3 failures in step 8.15.1_bwc-snapshots (42.9% fail rate in 7 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (40.0% fail rate in 5 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -160,9 +160,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.MlJobIT
   method: testDeleteJobAsync
   issue: https://github.com/elastic/elasticsearch/issues/112212
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=indices.create/20_synthetic_source/stored field under object with store_array_source}
-  issue: https://github.com/elastic/elasticsearch/issues/112264
 - class: org.elasticsearch.search.query.ScriptScoreQueryTests
   method: testScriptTermStatsAvailable
   issue: https://github.com/elastic/elasticsearch/issues/112278

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.create/20_synthetic_source.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.create/20_synthetic_source.yml
@@ -1342,8 +1342,8 @@ subobjects auto:
 # 112156
 stored field under object with store_array_source:
   - requires:
-      cluster_features: ["mapper.track_ignored_source"]
-      reason: requires tracking ignored source
+      cluster_features: ["mapper.source.synthetic_source_stored_fields_advance_fix"]
+      reason: requires bug fix to be implemented
 
   - do:
       indices.create:

--- a/server/src/main/java/org/elasticsearch/index/mapper/MapperFeatures.java
+++ b/server/src/main/java/org/elasticsearch/index/mapper/MapperFeatures.java
@@ -32,7 +32,8 @@ public class MapperFeatures implements FeatureSpecification {
             IndexModeFieldMapper.QUERYING_INDEX_MODE,
             NodeMappingStats.SEGMENT_LEVEL_FIELDS_STATS,
             BooleanFieldMapper.BOOLEAN_DIMENSION,
-            ObjectMapper.SUBOBJECTS_AUTO
+            ObjectMapper.SUBOBJECTS_AUTO,
+            SourceFieldMapper.SYNTHETIC_SOURCE_STORED_FIELDS_ADVANCE_FIX
         );
     }
 }

--- a/server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java
+++ b/server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java
@@ -38,7 +38,9 @@ import java.util.Locale;
 
 public class SourceFieldMapper extends MetadataFieldMapper {
     public static final NodeFeature SYNTHETIC_SOURCE_FALLBACK = new NodeFeature("mapper.source.synthetic_source_fallback");
-    public static final NodeFeature SYNTHETIC_SOURCE_STORED_FIELDS_ADVANCE_FIX = new NodeFeature("mapper.source.synthetic_source_stored_fields_advance_fix");
+    public static final NodeFeature SYNTHETIC_SOURCE_STORED_FIELDS_ADVANCE_FIX = new NodeFeature(
+        "mapper.source.synthetic_source_stored_fields_advance_fix"
+    );
 
     public static final String NAME = "_source";
     public static final String RECOVERY_SOURCE_NAME = "_recovery_source";

--- a/server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java
+++ b/server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java
@@ -38,6 +38,7 @@ import java.util.Locale;
 
 public class SourceFieldMapper extends MetadataFieldMapper {
     public static final NodeFeature SYNTHETIC_SOURCE_FALLBACK = new NodeFeature("mapper.source.synthetic_source_fallback");
+    public static final NodeFeature SYNTHETIC_SOURCE_STORED_FIELDS_ADVANCE_FIX = new NodeFeature("mapper.source.synthetic_source_stored_fields_advance_fix");
 
     public static final String NAME = "_source";
     public static final String RECOVERY_SOURCE_NAME = "_recovery_source";

